sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -56,18 +56,29 @@ export interface CreateRoleAssignmentOptions extends coreHttp.OperationOptions {
 export interface DeleteRoleAssignmentOptions extends coreHttp.OperationOptions {
 }
 
+// @public
+export interface DeleteRoleDefinitionOptions extends coreHttp.OperationOptions {
+}
+
 // @public
 export interface GetRoleAssignmentOptions extends coreHttp.OperationOptions {
 }
 
+// @public
+export interface GetRoleDefinitionOptions extends coreHttp.OperationOptions {
+}
+
 // @public
 export class KeyVaultAccessControlClient {
     constructor(vaultUrl: string, credential: TokenCredential, options?: AccessControlClientOptions);
     createRoleAssignment(roleScope: RoleAssignmentScope, name: string, roleDefinitionId: string, principalId: string, options?: CreateRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
     deleteRoleAssignment(roleScope: RoleAssignmentScope, name: string, options?: DeleteRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    deleteRoleDefinition(roleScope: RoleScope, name: string, options?: DeleteRoleDefinitionOptions): Promise<KeyVaultRoleDefinition>;
     getRoleAssignment(roleScope: RoleAssignmentScope, name: string, options?: GetRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    getRoleDefinition(roleScope: RoleScope, name: string, options?: GetRoleDefinitionOptions): Promise<KeyVaultRoleDefinition>;
     listRoleAssignments(roleScope: RoleAssignmentScope, options?: ListRoleAssignmentsOptions): PagedAsyncIterableIterator<KeyVaultRoleAssignment>;
     listRoleDefinitions(roleScope: RoleAssignmentScope, options?: ListRoleDefinitionsOptions): PagedAsyncIterableIterator<KeyVaultRoleDefinition>;
+    upsertRoleDefinition(roleScope: RoleScope, name: string, permissions: KeyVaultPermission[], description?: string, options?: UpsertRoleDefinitionOptions): Promise<KeyVaultRoleDefinition>;
```

Comment:
@bterlson @xirzec I was wondering what you think about this naming? According to our guidelines that's the recommendation right? Mind just confirming if you prefer this or `createOrUpdateRoleDefinition`? https://azure.github.io/azure-sdk/typescript_design.html#ts-client-naming-conventions

There's also an ongoing discussion about the terminology in dotnet https://github.com/Azure/azure-sdk-for-net/pull/18227#discussion_r566468667 so I figured it's probably better to double check 👍

Reply:
I agree upsert is the current guidelines and cosmos uses this (e.g. [here](https://docs.microsoft.com/en-us/javascript/api/@azure/cosmos/items?view=azure-node-latest#upsert_any__RequestOptions_)). I can't say I'm the biggest fan however.

Reply:
Citing Heath in a separate channel:

> In .NET for track 2 libraries (src only - no tests) there are 215 instances of CreateOrUpdate and only 42 of Upsert.

Reply:
Thanks @bterlson! 

I think I'm going to go with the guideline until told otherwise. Also agree that I'm not crazy about `upsert` but there's nothing that fits this case better than `upsert` in the list (and `createOrUpdate` is specifically listed as a bad example).

Reply:
I decided to go with CreateOrUpdate for now. We can chat in the next stand up about what to standardize on x-lang.